### ISSUE TITLE
New observation update problem

### ISSUE THREAD

**Issue by yenchenlin** [songrotek/DRL-FlappyBird#1]
Hello @songrotek , code [here](https://github.com/songrotek/DRL-FlappyBird/blob/master/BrainDQN_Nature.py#L137) seems to keep the oldest 3 frames forever, which means the algorithm is not using the newest 4 frames to represent state.

**Comment by floodsung** [comment]
it seems that 
`newState = np.append(self.currentState[:,:,1:],nextObservation,axis = 2)` 
is better

**Comment by yenchenlin** [comment]
Yes 👍

**Comment by qiaofei32** [comment]
![image](https://cloud.githubusercontent.com/assets/10598332/17281984/49614b24-57d3-11e6-8264-a465b608a183.png)
as you described in the picture，your code should as below:
`	def createQNetwork(self):
		# network weights
		W_conv1 = self.weight_variable([8,8,4,32])
		b_conv1 = self.bias_variable([32])

		W_conv2 = self.weight_variable([4,4,32,64])
		b_conv2 = self.bias_variable([64])

		W_conv3 = self.weight_variable([3,3,64,64])
		b_conv3 = self.bias_variable([64])

		# W_fc1 = self.weight_variable([1600,512]) # 1600 = 5*5*64 , 5*5 patch, 64 features when delete line 82 (1) and line 85 (2)
		W_fc1 = self.weight_variable([2*2*64,512]) # 		2*2*64 , 2*2 patch, 64 features 
		
		b_fc1 = self.bias_variable([512])

		W_fc2 = self.weight_variable([512,self.actions])
		b_fc2 = self.bias_variable([self.actions])

		# input layer
		stateInput = tf.placeholder("float",[None,80,80,4])

		# hidden layers
		h_conv1 = tf.nn.relu(self.conv2d(stateInput,W_conv1,4) + b_conv1)
		h_pool1 = self.max_pool_2x2(h_conv1)

		h_conv2 = tf.nn.relu(self.conv2d(h_pool1,W_conv2,2) + b_conv2)
		h_pool2 = self.max_pool_2x2(h_conv2) # max_pool with win: 2*2 --------------(1)

		h_conv3 = tf.nn.relu(self.conv2d(h_pool2,W_conv3,1) + b_conv3)
		h_pool3 = self.max_pool_2x2(h_conv3) # max_pool with win: 2*2 --------------(2)

		# h_conv3_flat = tf.reshape(h_pool3,[-1,1600])
		h_conv3_flat = tf.reshape(h_pool3,[-1,2*2*64]) # modified the size ---------(3)
		
		h_fc1 = tf.nn.relu(tf.matmul(h_conv3_flat,W_fc1) + b_fc1)

		# Q Value layer
		QValue = tf.matmul(h_fc1,W_fc2) + b_fc2

		return stateInput,QValue,W_conv1,b_conv1,W_conv2,b_conv2,W_conv3,b_conv3,W_fc1,b_fc1,W_fc2,b_fc2`
